tests/e2e/weaviate_e2e/test_weaviate_e2e.py

```diff
@@ -22,7 +22,7 @@
 from neo4j_genai.embedder import Embedder
 from neo4j_genai.retrievers import WeaviateNeo4jRetriever
 from neo4j_genai.types import RetrieverResult, RetrieverResultItem
-from weaviate.client import Client
+from weaviate.client import WeaviateClient
```

Comment:
Moved to using WeaviateClient due to v4 migration, see

https://weaviate.io/developers/weaviate/client-libraries/python/v3_v4_migration
https://github.com/run-llama/llama_index/pull/13719/files
This was causing mypy errors